### PR TITLE
rpcdaemon: read block number by transaction hash

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkworm/silkrpc/ethbackend/backend.hpp [hunk 1]
```diff
@@ -47,6 +47,7 @@ class BackEnd {
     virtual Task<NodeInfos> engine_node_info() = 0;
     virtual Task<PeerInfos> peers() = 0;
     virtual Task<bool> get_block(uint64_t block_number, const HashAsSpan& hash, bool read_senders, silkworm::Block& block) = 0;
+    virtual Task<uint64_t> get_block_number_from_txn_hash(const HashAsSpan& hash) = 0;
```
Comment (canepat) [inline silkworm/silkrpc/ethbackend/backend.hpp:50]: Please use `BlockNum` instead of plain `uint64_t` as block number type

**Review by canepat — APPROVED** [review]

**Review by canepat — CHANGES_REQUESTED** [review]
See comments above

File: silkworm/silkrpc/storage/chain_storage.hpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ class ChainStorage {
     //! Read total difficulty for block specified by hash and number
     [[nodiscard]] virtual Task<std::optional<intx::uint256>> read_total_difficulty(const Hash& block_hash, BlockNum block_number) const = 0;
 
-    // Task<uint64_t> read_block_number_by_transaction_hash(const evmc::bytes32& transaction_hash);
+    virtual Task<bool> read_block_number_by_transaction_hash(const evmc::bytes32& transaction_hash, uint64_t& bn) const = 0;
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/storage/chain_storage.hpp:95]: ok

File: silkworm/silkrpc/ethbackend/backend.hpp (hunk 1 above)
Comment (lupin012) [reply silkworm/silkrpc/ethbackend/backend.hpp:50]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**